sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/maskExpressions.scala

```diff
@@ -84,7 +86,7 @@ object MaskExpressionBuilder extends ExpressionBuilder {
     val digitCharArg = InputParameter("digitChar", Some(Literal(Mask.MASKED_DIGIT)))
     val otherCharArg = InputParameter(
       "otherChar",
-      Some(Literal(Mask.MASKED_IGNORE, StringType)))
+      Some(Literal(Mask.MASKED_IGNORE, SQLConf.get.defaultStringType)))
```

Comment:
why do we use the default string type as the expected input type here?

Reply:
`MASKED_IGNORE` is null, so we have to specify a type for this literal in func signature
so instead of `StringType(0)` it should only be correct to get the default string type

we enforce implicit cast for this expression, so in a situation where the first parameter takes the default string type (for example when not using explicit collation) there would be a conflict with this `StringType(0)` if it were to stay here

Reply:
Why are we then leaving digitCharArg as StringType? Either they all take StringType or SQLConf.get.defaultStringType.

Reply:
update: I fixed this in the latest commit, but I don't like the idea of doing this manually

instead, I think we should consider updating `literals.scala` to construct string literals with respect to `defaultStringType` instead of `StringType`

if that sound good @cloud-fan, then I'll get to that a separate effort

Reply:
We did this before but reverted. `Literal` is a low-level thing and it's better to not access SQLConf there.